### PR TITLE
Per Project Settings

### PR DESCRIPTION
**Pull request by spAnser**
Per project settings that adds to the existing global settings.

Supports multiple projects in single window.

### CODE REVIEW

**Comment by spAnser** [comment]
I just noticed a bug when creating a new file with this that needs to be addressed.

**Comment by spAnser** [comment]
Should be all good now.

File: lib/flex-tool-bar.coffee [hunk 1]
```diff
@@ -72,6 +77,25 @@ module.exports =
...
+  resolveProjectConfigPath: ->
+    editor = atom.workspace.getActiveTextEditor()
+
+    if editor?.buffer?.file?.getParent()?.path?
+      projectCount = editor.project.rootDirectories.length
+      count = 0
+      while count < projectCount
+        console.log count
+        pathToCheck = editor.project.rootDirectories[count].path
+        console.log pathToCheck
```
Comment (jerone) [inline lib/flex-tool-bar.coffee]: :fire: `console.log`

**Comment by spAnser** [comment]
Thought I already did a search for those :cry:

**Comment by jerone** [comment]
Did a primary review and looking interesting. Don't have much time.

Bit concerned about not disposing the watches.

**Comment by spAnser** [comment]
I left the watch in because I swap between a few projects in the same window and I wasn't sure if I should watch / unwatch each time I switch project files. I did make sure I only watch a project once though.

File: lib/flex-tool-bar.coffee [hunk 2]
```diff
@@ -87,6 +111,18 @@ module.exports =
       watch @configFilePath, =>
         @reloadToolbar(true)
 
+  registerProjectWatch: ->
+      if @projectToolbarConfigPath and jQuery.inArray(@projectToolbarConfigPath, @watchList) < 0
```
Comment (cakecatz) [inline lib/flex-tool-bar.coffee]: You should use `Array.prototype.indexOf` instead of `jQuery.inArray`.

**Comment by jerone** [comment]
Could you update the readme how to use this feature.

**Comment by jerone** [comment]
Besides that `node-watch` isn't added to the package.json file, I prefer to use the more popular [chokidar](https://www.npmjs.com/package/chokidar).

**Comment by cakecatz** [comment]
LGTM

**Comment by jerone** [comment]
Source looks good :+1:  I did not have time to test this. @cakecatz did you have time to test it?

**Comment by jerone** [comment]
This closes https://github.com/cakecatz/flex-toolbar/issues/42

**Comment by cakecatz** [comment]
@jerone Yeah. I'm using this feature for testing now.
If it's ok, I will merge this soon.

**Comment by jerone** [comment]
:shipit:

**Comment by cakecatz** [comment]
@spAnser Nice work :+1:

**Comment by spAnser** [comment]
Thanks recently found this and other plugins have used this type of feature. I thought it could be good for this one as well and saw the Issue with others wanting it and a help wanted tag.

**Pull request merged**